docs/core/development/policy/coding-standard.rst

```diff
@@ -625,6 +625,126 @@ Any attribute, method argument, or method name that corresponds *directly* to a
+
+All new code emitting log messages should use the new logging system.
+
+This section is dedicated to using the new logging system in Twisted core code.
+For general information about using the logging system please check the dedicated documentation about :doc:`how to log with twisted.logger </core/howto/logger>` and the :api:`twisted.logger.Logger <Logger>` API docs.
+
+General usage rules:
+
+* Don't use explicit namespaces.
+  The logger will automatically fill the namespace.
+* Use the global log publisher.
+  Twisted code may call into non-Twisted code which makes stdlib/legacy log messages, which then get redirected to the global logger.
+* When possible instantiate the Logger as ``self._log`` instance member.
+* Emitted logs are not part of the API/contract so you don't have to document it as part of the API.
+* For security features you might documented the emitted logs and then they will became part of the contract.
+
+Example of usage:
+
+.. code-block:: python
+
+    from twisted.logger import Logger
+
+    class SomeStuff(object):
+        """
+        Class using the new logging system as an instance member
+        """
+
+        _log = Logger()
+
+        def getUser(self, name):
+            try:
+                return self._otherStuff()
+            except Exception as error:
+                self._log.info(
+                    "Unable to check {user!s}", user=name, error=error)
+
+                raise UnauthorizedLogin()
+
+
+General testing rules:
+
+* New code calling the new logging system will test that the log is emitted.
+* You don't need to explicitly check all members of the structured log.
+* Emitted logs which are documented should have detailed tests which check all the conditions described in the documentation.
+
+
+Example of testing the code:
+
+.. code-block:: python
+
+    def test_getUserNonexistentDatabase(self):
+        """
+        A missing db file will cause a permanent rejection of authorization
+        attempts.
+        """
+        self.db = checkers.FilePasswordDB('test_thisbetternoteverexist.db')
+
+        with self.captureLogs() as logs:
+            self.failUnlessRaises(
+            error.UnauthorizedLogin, self.db.getUser, 'some-user')
+
+        self.assertIn('Unable to check some-user', logs.asText)
+        events = logs.asEvents
+        self.assertEqual(1, len(events))
+        self.assertEqual(events[0]['user'], 'some-user')
+        self.assertIsInstance(events[0]['error'] , error.UnauthorizedLogin)
+        self.assertEqual(events['log_level'], logs.LogLevel.info)
+
+
+Working around `self._log` conflicts.
+For older code you might now be able to use `self._log` as it can be already used for other purpose.
+In this case you can just use a different name for the new logging object or
+use the :api:`twisted.logger._logger._loggerFor <_loggerFor>` helper.
+
+.. code-block:: python
+
+    from twisted.logger._logger import _loggerFor
+
+    class SomeStuff(object):
+        """
```

Comment:
Again I think we should dispense with the docstrings for example code.  It's already not quite twistedchecker-friendly (no `@ivar`s), so it's not verbose enough to be realistic.